source/common/http/conn_manager_utility.cc

```diff
@@ -77,13 +77,27 @@ Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequest
     }
     request_headers.insertForwardedProto().value().setReference(
         connection.ssl() ? Headers::get().SchemeValues.Https : Headers::get().SchemeValues.Http);
+
+    // Since useRemoteAddress() is true, skip adding x-forwarded-port when the x-forwarded-port
```

Comment:
I think this logic should be inverted? If remote address is true, we are taking the immediate peer as the remote, so I think we should overwrite x-fowarded-port if it exists (or append)?